platform/darwin/docs/guides/Style Layers Using Expressions.md.ejs

```diff
@@ -0,0 +1,168 @@
+
+## Stops
+
+tops are dictionary keys that are associated with layer attribute values. You can use feature attribute values, zoom levels, and heatmap point density as keys, with an constant value or expression for the value. For example, you can use a stop dictionary with the zoom levels 0, 10, and 20 as keys and the colors yellow, orange, and red as the values.
+<%- guideExample(guide, 'Stops', os) %>
+
+Matched values can be used in the place of stops dictionaries with categorical interpolation. The `MGL_MATCH()` custom function creates feature and layer attribute value pairs.
+## Interpolation mode
+
+The effect a key has on the layer style value is determined by the interpolation mode. There are four interpolation modes that can be used with a style expression: exponential/linear, interval, categorical, and identity. You can also use exponential/linear and categorical interpolation modes with a camera expression.
+### Linear
+
+`MGLExpressionInterpolationModeLinear` interpolates linearly between stop values.
+The stops dictionary below, for example, shows colors that continuously shift from yellow to orange to red to blue to white based on the attribute value.
+<%- guideExample(guide, 'Linear', os) %>
+
+![exponential mode](img/data-driven-styling/linear.png)
+
+### Exponential
+
+By combining `MGLExpressionInterpolationModeExponential` with an interpolation base greater than `0` (other than `1`), you can interpolate between values exponentially, create an accelerated ramp effect.
+Here’s a visualization from Mapbox Studio (see [Working with Mapbox Studio](working-with-mapbox-studio.html)) comparing interpolation base values of `1.5` and `0.5` based on zoom.
+
+<img src="img/data-driven-styling/exponential-function.png" height=344/>
+<img src="img/data-driven-styling/exponential-function-1.png" height=344/>
+
+The example below increases a layer’s `circleRadius` exponentially based on a map’s zoom level. The interpolation base is `1.5`.
+<%- guideExample(guide, 'Exponential', os) %>
+
+### Interval
+
+Steps, or intervals, create a range using the keys from the stops dictionary. The range is from the given key to just less than the next key. The attribute values that fall into that range are then styled using the layout or paint value assigned to that key.
+
+When we use the stops dictionary given above with an interval interpolation mode, we create ranges where earthquakes with a magnitude of 0 to just less than 2.5 would be yellow, 2.5 to just less than 5 would be orange, and so on.
+<%- guideExample(guide, 'Interval', os) %>
+
+![interval mode](img/data-driven-styling/interval.png)
+
+### Categorical
+
+At each stop, a categorical, or match expression, produces an output value equal to the function input. We’re going to match different feature attribute and style layer property values than we did for the previous two modes.
```

Comment:
I think it’s confusing to explain each of the new functions in detail while using the old terminology. Ideally, “Predicates and Expressions” would explain what `match` means, along with the detailed example. Meanwhile, this guide would offer a before-and-after of the old v3._x_ code and the corresponding v4.0 code, along with a minimal explanation, like this:

> A function with categorical interpolation corresponds to a match expression. Use the `MGL_MATCH()` syntax to associate specific input values with output values.
> 
> [Before: code block containing style function with categorical interpolation]
> [After: code block containing match expression]
> 
> `MGL_MATCH` takes the attribute key, followed by possible attribute and style value pairs. The final argument is a default style value.